File: Array/209_minimum_size_subarray_sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        left = -1
        n = len(nums)
        min_len = n + 1
        prefix_sum = 0

        for right in range(n):
            if nums[right] >= target:
                return 1
            prefix_sum += nums[right]
            if prefix_sum >= target:
                while left < right and prefix_sum >= target:
                    min_len = min(min_len, right - left)
                    left += 1
                    prefix_sum -= nums[left]

        return min_len if min_len != n + 1 else 0

File: Array/209_minimum_size_subarray_sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int prefix_sum = 0, left = 0, min_length = INT_MAX;

        for(int right = 0; right < nums.size(); right += 1) {
            if(nums[right] >= target) return 1;

            prefix_sum += nums[right];
            while(prefix_sum >= target) {
                min_length = min(min_length, right - left + 1);
                prefix_sum -= nums[left++];
            }
        }

        return min_length != INT_MAX ? min_length : 0;
    }
};

